### PR TITLE
More refactoring for GC evaluations

### PR DESCRIPTION
**Pull request by jeromekelleher**
- Rename gc data file
- Update the GC code to validate the parameters.

### CODE REVIEW

**Comment by jeromekelleher** [comment]
I've done some tidying up here @fbaumdicker and added some code to automate the validation. You're right about fastsimbac, it's pretty weird the difference in distribution:

verify_fastsimbac_v_msprime

As opposed to the corresponding number of trees (==breakpoints) for simbac/msprime:

verify_simbac_v_msprime

**Comment by jeromekelleher** [comment]
Huh, fastSimBac is also *way* slower than SimBac, despite generating substantially fewer trees.

Any thoughts on what's going on here @fbaumdicker?

**Comment by jeromekelleher** [comment]
I guess we should try to merge this much first before going any further with the fastSimBac stuff. @fbaumdicker, would you mind having a quick look through the code to see if you think it's looking OK please?

File: evaluation/generate_gc_perf_data.py [hunk 1]
```diff
@@ -1,114 +1,192 @@
...
+                gc_tract_length=gc_tract_length,
+                count_trees=True,
+            )
+            nt_fastsimbac[j] = run_fastsimbac(
+                sample_size=sample_size,
+                L=L,
+                gc_rate=gc_rate,
+                gc_tract_length=gc_tract_length,
+                count_trees=True,
+            )
```
Comment (fbaumdicker) [inline evaluation/generate_gc_perf_data.py:108]:
```suggestion
            nt_fastsimbac[j] = run_fastsimbac(
                sample_size=sample_size,
                L=L,
                gc_rate=gc_rate,
                gc_tract_length=gc_tract_length,
                set_seed = j,
                count_trees=True,
            )
```

File: evaluation/generate_gc_perf_data.py [hunk 2]
```diff
@@ -1,114 +1,192 @@
...
+def run_fastsimbac(*, sample_size, L, gc_rate, gc_tract_length, count_trees=False):
+
+    # using R=2*gc_rate as gene conversion/recombination rate as SimBac uses R/2
+    R = gc_rate * 2
+    # Set theta to 0 to remove mutations (defaults to 0.01)
+    args = f"{sample_size} {int(L)} -r {R} {gc_tract_length} -t 0"
+    if count_trees:
+        args += " -T "
+
+    # print("running", args)
```
Comment (fbaumdicker) [inline evaluation/generate_gc_perf_data.py:48]:
```suggestion
    if count_trees:
        args += " -T "
    if set_seed > 0:
        args += " -s " + str(set_seed)

    # print("running", args)
```

File: evaluation/generate_gc_perf_data.py [hunk 3]
```diff
@@ -1,114 +1,192 @@
...
+    num_trees = 0
+    if count_trees:
+        with open(outfile) as f:
+            for line in f:
+                num_trees += 1
+
+    return num_trees
+
+
+def run_fastsimbac(*, sample_size, L, gc_rate, gc_tract_length, count_trees=False):
```
Comment (fbaumdicker) [inline evaluation/generate_gc_perf_data.py:39]:
```suggestion
def run_fastsimbac(*, sample_size, L, gc_rate, gc_tract_length, set_seed=0, count_trees=False):
```

**Review by fbaumdicker — COMMENTED** [review]
At least the strange horizontal lines disappear if you explicitly set the used seeds in fastSimBac. Otherwise, for very quick runs the same seed will be used multiple times.

However, this does still not solve the gc_rate vs. number of trees issue.

**Comment by fbaumdicker** [comment]
Nice work @jeromekelleher. I added some suggestions to use different seeds. If different seeds are used the figure looks like this:

verify_fastsimbac_v_msprime


So still not the correct number of trees, but at least no strange distribution.

Regarding the running time of fastSimBac, I would guess that fastSimBac is only faster for long sequences.
But I am not sure if this is true.

**Comment by jeromekelleher** [comment]
Thanks @fbaumdicker, great spot on the seeds issue! I'll play around a bit more with fastsimbac and see if I can make sense of it.

**Pull request merged**